File: front-end/demo-app/models/astronaut-exam-score.ts
import { Entity, IAggregateRoot, Required } from '../../src';

export class AstronautExamScore extends Entity implements IAggregateRoot {
    @Required() public examineeUserId: string;
    public passed: boolean;
    public score: number;

    constructor(options?: IAstronautExamScoreOptions) {
        options = options || {};
        super(options);
        this.examineeUserId = options!.examineeUserId;
        this.passed = options!.passed;
        this.score = options!.score;
    }
}

interface IAstronautExamScoreOptions {
    examineeUserId?: string;
    passed?: boolean;
    score?: number;
    id?: number;
}

File: TestDomain/AstronautExamScore.cs
using FluentNHibernate.Data;
using lvl.Ontology;
using lvl.Ontology.Authorization;
using System.ComponentModel.DataAnnotations;

namespace lvl.TestDomain
{
    [OwnedByUser(nameof(ExamineeUserId))]
    public class AstronautExamScore: Entity, IAggregateRoot
    {
        [Required]
        public string ExamineeUserId { get; set; }
        public bool Passed { get; set; }
        public decimal Score { get; set; }
    }
}
